build-aux/m4/bitcoin_qt.m4

```diff
@@ -372,16 +411,34 @@ dnl Outputs: All necessary QT_* variables are set.
 dnl Outputs: have_qt_test and have_qt_dbus are set (if applicable) to yes|no.
 AC_DEFUN([_BITCOIN_QT_FIND_LIBS],[
   BITCOIN_QT_CHECK([
-    PKG_CHECK_MODULES([QT_CORE], [${qt_lib_prefix}Core${qt_lib_suffix} $qt_version], [QT_INCLUDES="$QT_CORE_CFLAGS $QT_INCLUDES" QT_LIBS="$QT_CORE_LIBS $QT_LIBS"],
+    PKG_CHECK_MODULES([QT_CORE], [${qt_lib_prefix}Core${qt_lib_suffix} $qt_version], [],
                       [BITCOIN_QT_FAIL([${qt_lib_prefix}Core${qt_lib_suffix} $qt_version not found])])
+    if test "$bitcoin_qt_want_version" = "qt6"; then
+      QT_CORE_LIBS="$QT_CORE_LIBS -lm ${qt_lib_path}/libQt6BundledPcre2.a -ldl -lrt"
+    fi
+    QT_INCLUDES="$QT_CORE_CFLAGS $QT_INCLUDES"
+    QT_LIBS="$QT_CORE_LIBS $QT_LIBS"
+    _BITCOIN_QT_CHECK_APP([QCoreApplication])
   ])
   BITCOIN_QT_CHECK([
-    PKG_CHECK_MODULES([QT_GUI], [${qt_lib_prefix}Gui${qt_lib_suffix} $qt_version], [QT_INCLUDES="$QT_GUI_CFLAGS $QT_INCLUDES" QT_LIBS="$QT_GUI_LIBS $QT_LIBS"],
+    PKG_CHECK_MODULES([QT_GUI], [${qt_lib_prefix}Gui${qt_lib_suffix} $qt_version], [],
                       [BITCOIN_QT_FAIL([${qt_lib_prefix}Gui${qt_lib_suffix} $qt_version not found])])
+    if test "$bitcoin_qt_want_version" = "qt6"; then
+      QT_GUI_LIBS="$QT_GUI_LIBS ${qt_lib_path}/libQt6BundledLibpng.a ${qt_lib_path}/libQt6BundledHarfbuzz.a -lfreetype -lfontconfig -lxkbcommon"
+    fi
+    QT_INCLUDES="$QT_GUI_CFLAGS $QT_INCLUDES"
+    QT_LIBS="$QT_GUI_LIBS $QT_LIBS"
+    _BITCOIN_QT_CHECK_APP([QGuiApplication])
   ])
   BITCOIN_QT_CHECK([
-    PKG_CHECK_MODULES([QT_WIDGETS], [${qt_lib_prefix}Widgets${qt_lib_suffix} $qt_version], [QT_INCLUDES="$QT_WIDGETS_CFLAGS $QT_INCLUDES" QT_LIBS="$QT_WIDGETS_LIBS $QT_LIBS"],
+    PKG_CHECK_MODULES([QT_WIDGETS], [${qt_lib_prefix}Widgets${qt_lib_suffix} $qt_version], [],
                       [BITCOIN_QT_FAIL([${qt_lib_prefix}Widgets${qt_lib_suffix} $qt_version not found])])
+    if test "$bitcoin_qt_want_version" = "qt6"; then
+      QT_WIDGETS_LIBS="$QT_WIDGETS_LIBS ${qt_lib_path}/objects-Release/Widgets_resources_1/.rcc/qrc_qstyle.cpp.o ${qt_lib_path}/objects-Release/Widgets_resources_2/.rcc/qrc_qstyle1.cpp.o ${qt_lib_path}/objects-Release/Widgets_resources_3/.rcc/qrc_qmessagebox.cpp.o"
```

Comment:
In [9f61d6a7605eee4b116b8791276bd71baf75bf10](https://github.com/bitcoin/bitcoin/pull/24798/commits/9f61d6a7605eee4b116b8791276bd71baf75bf10):
Can you explain what's going on here? Trying to link against `.o` files in configure checks not only seems very fragile, but downright hacky.

Reply:
This is reproducing of lacking `pkg-config --static` functionality. List of files to link against has been taken from `*.prl` files.